File: org.kevoree.modeling.microframework.typescript/src/main/ts/org/kevoree/modeling/api/InboundReference.ts
///<reference path="meta/MetaReference.ts"/>

class InboundReference {

  private reference: MetaReference = null;
  private object: KObject<any,any> = null;

  constructor(reference: MetaReference, object: KObject<any,any>) {
    this.reference = reference;
    this.object = object;
  }

  public getReference(): MetaReference {
    return this.reference;
  }

  public getObject(): KObject<any,any> {
    return this.object;
  }

}

File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/InboundReference.java
package org.kevoree.modeling.api;

import org.kevoree.modeling.api.meta.MetaReference;

/**
 * Created by duke on 10/17/14.
 */
public class InboundReference {

    private MetaReference _reference;

    private long _source;

    public InboundReference(MetaReference p_reference, long p_source) {
        this._reference = p_reference;
        this._source = p_source;
    }

    public MetaReference reference() {
        return _reference;
    }

    public long source() {
        return _source;
    }

}
